src/data/polynomial/high_order_derivative.lean

```diff
@@ -0,0 +1,153 @@
+import data.polynomial.derivative
+import algebra.ring.basic
+import data.polynomial.basic
+import logic.function.iterate
+import tactic.ring tactic.linarith
+
+noncomputable theory
+
+open finset nat polynomial
+open_locale big_operators
+
+namespace polynomial
+universes u
+variable {R : Type u}
+
+section semiring
+
+variables [semiring R] (f p q: polynomial R) (n k : ℕ)
+
+/--Defining the n-th derivative of polynomial f-/
+def deriv_n : polynomial R := derivative ^[n] f
+
+@[simp] lemma zeroth_deriv : deriv_n f 0 = f := rfl
+
+
+lemma deriv_n_succ : (deriv_n f n).derivative = (deriv_n f (n+1)) :=
```

Comment:
I would expect this lemma to be the other way around, and the uses in this file seems to agree with what I think.

Reply:
So `(deriv_n f (n+1)) = (deriv_n f n).derivative`